File: src/api/services/dashboardService.js
import apiClient from '@/config/axios';

/**
 * Dashboard API service.
 *
 * All calls hit /api/v1/dashboard/* endpoints.
 */
const dashboardService = {
  /** Aggregated KPI counts (products, categories, warehouses, suppliers). */
  getStats: () => apiClient.get('/dashboard/stats'),

  /** Products whose currentStock ≤ reorderLevel. */
  getLowStock: () => apiClient.get('/dashboard/low-stock'),

  /** Products with currentStock ≤ 0. */
  getOutOfStock: () => apiClient.get('/dashboard/out-of-stock'),

  /** Top products by stock quantity. */
  getTopProducts: (limit = 6) => apiClient.get('/dashboard/top-products', { params: { limit } }),

  /** Detailed product statistics. */
  getProductStatistics: () => apiClient.get('/dashboard/product-statistics'),

  /** Inventory value breakdown by category. */
  getInventoryValue: () => apiClient.get('/dashboard/inventory-value'),

  /** Recent purchase orders for dashboard display. */
  getRecentOrders: (limit = 5) => apiClient.get('/dashboard/recent-orders', { params: { limit } }),
};

export default dashboardService;

File: backend/src/main/java/com/smartwms/service/DashboardService.java
package com.smartwms.service;

import com.smartwms.dto.response.DashboardPurchaseOrderResponse;
import com.smartwms.dto.response.DashboardStatsResponse;
import com.smartwms.dto.response.InventoryValueResponse;
import com.smartwms.dto.response.LowStockProductResponse;
import com.smartwms.dto.response.ProductStatisticsResponse;
import com.smartwms.dto.response.TopProductResponse;

import java.util.List;

/**
 * Service contract for dashboard aggregations and KPI metrics.
 */
public interface DashboardService {

    /**
     * Returns aggregated KPI counts for the main dashboard.
     */
    DashboardStatsResponse getStats();

    /**
     * Returns products whose current stock is at or below the reorder level.
     */
    List<LowStockProductResponse> getLowStockProducts();

    /**
     * Returns products that are completely out of stock.
     */
    List<LowStockProductResponse> getOutOfStockProducts();

    /**
     * Returns top products by stock quantity.
     */
    List<TopProductResponse> getTopProducts(int limit);

    /**
     * Returns detailed product statistics.
     */
    ProductStatisticsResponse getProductStatistics();

    /**
     * Returns inventory value breakdown by category.
     */
    InventoryValueResponse getInventoryValue();

    /**
     * Returns recent purchase orders for dashboard display.
     */
    List<DashboardPurchaseOrderResponse> getRecentOrders(int limit);
}
